### PR TITLE
Improve error handling when widgets configured

### PR DESCRIPTION
**Pull request by elParaguayo**
Rather than crashing qtile completely, the bar will now replace a broken widget with a warning for the user.

This is messy/proof of concept* so interested in thoughts here.

*There will be errors in tests.

### CODE REVIEW

**Comment by tych0** [comment]
Any chance you can add a test for this?

File: libqtile/bar.py [hunk 1]
```diff
@@ -201,8 +203,18 @@ def _configure(self, qtile, screen):
 
         if self.window:
             self.window.place(self.x, self.y, self.width, self.height, 0, None)
+            crashed_widgets = []
             for i in self.widgets:
-                i._configure(qtile, self)
+                try:
```
Comment (elParaguayo) [inline libqtile/bar.py:208]: There's some duplication here and below. Will look to refactor.

**Comment by tych0** [comment]
Thanks!

**Comment by tych0** [comment]
Hrm, seems like something about this made CI unhappy?

**Comment by elParaguayo** [comment]
Really? Tests all ran ok when I pushed it.

Do you want to revert it and I can take a look?

**Comment by tych0** [comment]
Yeah, although I merged a bunch of stuff all before this one, so it's probably the interactions between separate PRs. It happens, I'm looking at it now. If I can't figure it out shortly I'll revert. The tests seem to be running fine (although I'm only 50% of the way through) with things reverted, so it's only this one patch.

**Comment by elParaguayo** [comment]
No problem. I can rebase onto latest master and rework it.

**Comment by elParaguayo** [comment]
So this will be fixed by #2243 so nothing more to do here.

**Pull request merged**